tests/testthat/test-cubes.R

```diff
@@ -218,20 +218,40 @@ with_test_authentication({
                 )
             )
         )
-        expect_equivalent(
-            as.array(crtabs(~v8 + v7, data = ds, useNA = "always")),
-            array(c(
-                5, 5,
-                3, 2,
-                2, 3,
-                0, 0
-            ),
-            dim = c(2L, 4L),
-            dimnames = list(
-                v8 = c("1955-11-05", "1955-11-06"),
-                v7 = c(LETTERS[3:5], "No Data")
-            )
-            )
+        expect_true(
+            isTRUE(all.equal(
+                as.array(crtabs(~v8 + v7, data = ds, useNA = "always")),
+                array(c(
+                    5, 5, 0,
+                    3, 2, 0,
+                    2, 3, 0,
+                    0, 0, 0
+                ),
+                dim = c(3L, 4L),
+                dimnames = list(
+                    v8 = c("1955-11-05", "1955-11-06", "<NA>"),
+                    v7 = c(LETTERS[3:5], "No Data")
+                )
+                ),
+                check.attributes = FALSE
+            ))
+            # Legacy output, if "No Data" categories are not automatically added:
+            || isTRUE(all.equal(
+                as.array(crtabs(~v8 + v7, data = ds, useNA = "always")),
+                array(c(
+                    5, 5,
+                    3, 2,
+                    2, 3,
+                    0, 0
+                ),
+                dim = c(2L, 4L),
+                dimnames = list(
+                    v8 = c("1955-11-05", "1955-11-06"),
+                    v7 = c(LETTERS[3:5], "No Data")
+                )
+                ),
+                check.attributes = FALSE
+            ))
```

Comment:
Would it be too much to have the eventually-will-be-true test alongside here so we don't have to swim through git history to remember how we constructed the `expect_equivalent` (and that we used `equivalent`). Even just commenting in the `expect_equivalent(...` inline with the non-legacy test here would be 💯

Reply:
I added comments to all the complicated ones describing what to do when the other ticket goes.